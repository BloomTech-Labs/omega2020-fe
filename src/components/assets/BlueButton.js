import React from 'react';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import { blue } from '@material-ui/core/colors';
import { withStyles } from '@material-ui/core/styles';

const BlueButton = (props) => {
  return (
    <ColorButton
      type={props.type}
      href={props.href}
      size='large'
      className='href'
      onClick={props.click}
    >
      <Typography variant='body2' className='title'>
        {props.title}
      </Typography>
    </ColorButton>
  );
};

const ColorButton = withStyles((theme) => ({
  root: {
    paddingLeft: 60,
    paddingRight: 60,
    paddingTop: 15,
    paddingBottom: 15,
    color: theme.palette.getContrastText(blue.A700),
    backgroundColor: blue.A700,
    '&:hover': {
      backgroundColor: blue[800],
    },
  },
}))(Button);

export default BlueButton;
