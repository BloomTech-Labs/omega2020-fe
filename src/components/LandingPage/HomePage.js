import React from 'react';
import useMediaQuery from '@material-ui/core/useMediaQuery';
import Box from '@material-ui/core/Box';
import Skeleton from '@material-ui/lab/Skeleton';
import CardHeader from '@material-ui/core/CardHeader';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import Tooltip from '@material-ui/core/Tooltip';
import { grey } from '@material-ui/core/colors';
import { makeStyles } from '@material-ui/core/styles';

import CategoryIcon from '@material-ui/icons/Category';
import AppsIcon from '@material-ui/icons/Apps';
import GetAppIcon from '@material-ui/icons/GetApp';

import BlueButton from '../assets/BlueButton';
import Cube from '../../images/cube4.gif';

const HomePage = (props) => {
  const classes = useStyles();
  const { loading = false } = props;

  let viewPort = false;
  const md = useMediaQuery('(min-width: 1050px)');
  if (viewPort === md) {
    viewPort = true;
  } else {
    viewPort = false;
  }

  return (
    <Box className={classes.root}>
      <Box className={classes.mainContainer}>
        <CardHeader
          avatar={
            loading ? (
              <Skeleton
                className='bannerIMG'
                variant='rect'
                width={600}
                height={600}
              />
            ) : (
              <Box className={classes.headerBox}>
                {(() => {
                  switch (viewPort) {
                    case true:
                      return (
                        <Box>
                          <Typography
                            className={classes.header}
                            style={{ fontSize: '1.3rem' }}
                            id='sudo'
                          >
                            Sudomega Interactive Sudoku
                          </Typography>
                          <br />
                          <Typography className={classes.subheader} id='upload'>
                            Learn how to play, upload your own sudoku
                            <br />
                            games, and compete with friends :&#41;
                          </Typography>
                        </Box>
                      );
                    default:
                      return (
                        <Box>
                          <Typography variant='h3' className={classes.header}>
                            Sudomega Interactive Sudoku
                          </Typography>
                          <br />
                          <Typography
                            variant='body1'
                            className={classes.subheader}
                          >
                            Learn how to play, upload your own sudoku
                            <br />
                            games, and compete with friends :&#41;
                          </Typography>
                        </Box>
                      );
                  }
                })()}
              </Box>
            )
          }
        />
        <CardHeader
          action={
            loading ? (
              <Skeleton className={classes.text} width={60} height={55} />
            ) : (
              <Box>
                <br />
                <br />
                <Box className={classes.space}>
                  <BootstrapTooltip title='Get Started'>
                    <IconButton color='inherit' href={'/tutorial'}>
                      <CategoryIcon className={classes.icon} />
                    </IconButton>
                  </BootstrapTooltip>
                  <BootstrapTooltip title='Play'>
                    <IconButton color='inherit' href={'/levels'}>
                      <AppsIcon className={classes.icon} />
                    </IconButton>
                  </BootstrapTooltip>
                  <BootstrapTooltip title='Upload'>
                    <IconButton color='inherit' href={'/upload'}>
                      <GetAppIcon className={classes.icon} />
                    </IconButton>
                  </BootstrapTooltip>
                </Box>
                <br />
                <br />
              </Box>
            )
          }
        />
        <CardHeader
          action={
            loading ? (
              <Skeleton className={classes.text} width={60} height={55} />
            ) : (
              <Box>
                <BlueButton title={'Get Started'} href={'/tutorial'} />
              </Box>
            )
          }
        />

        <CardHeader
          action={
            loading ? (
              <Skeleton className={classes.text} width={60} height={55} />
            ) : (
              <Box className={classes.gameBoard}>
                {/* <RenderPuzzle /> */}
                <img src={Cube} alt={'cibe'} width={300} />
              </Box>
            )
          }
        />
      </Box>
    </Box>
  );
};

const useStylesBootstrap = makeStyles((theme) => ({
  arrow: {
    color: grey[200],
  },
  tooltip: {
    backgroundColor: grey[200],
    color: '#000',
  },
}));

function BootstrapTooltip(props) {
  const classes = useStylesBootstrap();

  return <Tooltip arrow classes={classes} {...props} />;
}

const useStyles = makeStyles((theme) => ({
  root: {
    width: '100%',
    display: 'flex',
    flexFlow: 'column wrap',
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 100,
    marginBotom: 100,
  },
  mainContainer: {
    display: 'flex',
    flexFlow: 'column wrap',
    justifyContent: 'center',
    alignItems: 'center',
    textAlign: 'center',
  },
  subheader: {
    // opacity: 0.3,
    // color: '#A2A2A2',
  },
  icon: {
    fontSize: 80,
    color: grey[700],
  },
  gameBoard: {
    marginTop: 100,
  },
  space: {
    '& > *': {
      margin: theme.spacing(1),
    },
  },
}));

export default HomePage;
