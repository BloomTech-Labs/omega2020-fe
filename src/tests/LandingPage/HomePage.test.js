import React from 'react';
import renderer from 'react-test-renderer';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import '@testing-library/jest-dom/extend-expect';
import { Typography } from '@material-ui/core';
import BlueButton from '../../components/assets/BlueButton';
import HomePage from '../../components/LandingPage/HomePage';

configure({ adapter: new Adapter() });

describe('the information on the banner', () => {
    const renderHome = renderer.create(<HomePage />); //r represent renderer
    const rootHome = renderHome.root; //i represent instance

    const wrapper = mount(<HomePage />);
    const iWrapper = wrapper.instance();
    
    it('snapshot the info on the homepage', () => {
        const Home = renderer.create(<HomePage />).toJSON();
        expect(Home).toMatchSnapshot();
    });

    it('checks the title prop in the BlueButton', () => {
        expect(rootHome.findByType(BlueButton).props.title).toBe('Get Started; learn the game play');
    });

    it('checks the href prop in the BlueButton', () => {
        expect(rootHome.findByType(BlueButton).props.href).toBe('/tutorial')
    })

  it('checks the root for info', () => {
    expect(wrapper.containsMatchingElement(
      <Typography> 
          Sudomega Interactive Sudoku
      </Typography>
    )).toBeTruthy();
    
    expect(wrapper.containsMatchingElement(
      <Typography>
          Learn how to play, upload your own sudoku
          <br />
          games, and compete with friends :&#41;
      </Typography>
    )).toBeTruthy();

    expect(wrapper.containsMatchingElement(
        <Typography>
            No signup required
        </Typography>
    )).toBeTruthy();


    expect(iWrapper).toEqual(null)
    expect(rootHome.findByProps({id: 'sudo'}).props.children).toBeTruthy();
    expect(rootHome.findByProps({id: 'upload'}).props.children).toBeTruthy();    

  });

});
