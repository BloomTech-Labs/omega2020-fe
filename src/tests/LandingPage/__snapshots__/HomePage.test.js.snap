// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the information on the banner snapshot the info on the homepage 1`] = `
<div
  className="MuiBox-root MuiBox-root-27 makeStyles-root-1"
>
  <div
    className="MuiBox-root MuiBox-root-28 makeStyles-mainContainer-2"
  >
    <div
      className="MuiCardHeader-root"
    >
      <div
        className="MuiCardHeader-avatar"
      >
        <div
          className="MuiBox-root MuiBox-root-29"
        >
          <div
            className="MuiBox-root MuiBox-root-30"
          >
            <p
              className="MuiTypography-root MuiTypography-body1"
              id="sudo"
              style={
                Object {
                  "fontSize": "1.3rem",
                }
              }
            >
              Sudomega Interactive Sudoku
            </p>
            <br />
            <p
              className="MuiTypography-root makeStyles-subheader-3 MuiTypography-body1"
              id="upload"
            >
              Learn how to play, upload your own sudoku
              <br />
              games, and compete with friends :)
            </p>
          </div>
        </div>
      </div>
      <div
        className="MuiCardHeader-content"
      />
    </div>
    <div
      className="MuiCardHeader-root"
    >
      <div
        className="MuiCardHeader-content"
      />
      <div
        className="MuiCardHeader-action"
      >
        <div
          className="MuiBox-root MuiBox-root-31"
        >
          <br />
          <br />
          <div
            className="MuiBox-root MuiBox-root-32 makeStyles-space-6"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root makeStyles-icon-4"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2l-5.5 9h11z"
              />
              <circle
                cx="17.5"
                cy="17.5"
                r="4.5"
              />
              <path
                d="M3 13.5h8v8H3z"
              />
            </svg>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root makeStyles-icon-4"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M4 8h4V4H4v4zm6 12h4v-4h-4v4zm-6 0h4v-4H4v4zm0-6h4v-4H4v4zm6 0h4v-4h-4v4zm6-10v4h4V4h-4zm-6 4h4V4h-4v4zm6 6h4v-4h-4v4zm0 6h4v-4h-4v4z"
              />
            </svg>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root makeStyles-icon-4"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 9h-4V3H9v6H5l7 7 7-7zM5 18v2h14v-2H5z"
              />
            </svg>
          </div>
          <br />
          <br />
        </div>
      </div>
    </div>
    <div
      className="MuiCardHeader-root"
    >
      <div
        className="MuiCardHeader-content"
      />
      <div
        className="MuiCardHeader-action"
      >
        <div
          className="MuiBox-root MuiBox-root-33"
        >
          <a
            aria-disabled={false}
            className="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-14 MuiButton-text href MuiButton-textSizeLarge MuiButton-sizeLarge"
            href="/tutorial"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
          >
            <span
              className="MuiButton-label"
            >
              <p
                className="MuiTypography-root title MuiTypography-body2"
              >
                Get Started; learn the game play
              </p>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </a>
          <br />
          <br />
          <span
            className="MuiTypography-root MuiTypography-caption"
          >
            No signup required
          </span>
        </div>
      </div>
    </div>
    <div
      className="MuiCardHeader-root"
    >
      <div
        className="MuiCardHeader-content"
      />
      <div
        className="MuiCardHeader-action"
      >
        <div
          className="MuiBox-root MuiBox-root-34 makeStyles-gameBoard-5"
        >
          <div
            className="makeStyles-root-16"
          >
            <div>
              <div
                className="makeStyles-board-17"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
